cores/esp8266/Arduino.h

```diff
@@ -165,6 +165,15 @@ void ets_intr_unlock();
 #define interrupts() xt_rsil(0)
 #define noInterrupts() xt_rsil(15)
 
+#define lockEnterVar(varSave, level) __asm__ __volatile__("rsil %0," __STRINGIFY(level) : "=a" (varSave))
```

Comment:
I think the code looks good, but I'm not sure why these `lock*` macros are in `Arduino.h`?  There is no documentation and they're unique to our setup, so I would not expect users to ever see or know how use them.  Maybe `Schedule.cpp` is the right place for them to live?

Also, and maybe this is @devyte getting in my head, would the locking mechanism make sense to be put into some sort of object?  On creation it disables IRQs and on deletion it resets IRQs.  Then you simply instantiate a InterruptLock() in your code block and it automagically does the save/restore when it is created and its lifetime ends.  I know I've seen something similar in other systems...it's a minor niggle and not something to stop the PR, though.

Reply:
Oh cool, the inception works! :laughing:
It's a good idea to make it an object that locks in the constructor and unlocks in the destructor. This makes sure that the lock is always released on scope exit, whether via normal way or exception, and it also makes for cleaner code when there are multiple returns due to error conditions or whatever.

Reply:
This is brilliant !